i6_models/parts/conformer/frontend.py

```diff
@@ -0,0 +1,279 @@
+from __future__ import annotations
+
+__all__ = [
+    "ConformerVGGFrontendV1Config",
+    "ConformerVGGFrontendV1",
+    "ConformerVGGFrontendV2Config",
+    "ConformerVGGFrontendV2",
+]
+
+from dataclasses import dataclass
+from typing import Callable, Optional, Tuple, Union
+
+import torch
+from torch import nn
+
+from i6_models.config import ModelConfiguration
+
+
+@dataclass
+class ConformerVGGFrontendV1Config(ModelConfiguration):
+    """
+    Attributes:
+        in_features: feature dimension of input
+        conv1_channels: number of channels for first conv layers
+        conv2_channels: number of channels for second conv layers
+        conv3_channels: number of channels for third conv layers
+        conv4_channels: number of channels for fourth dconv layers
+        conv_kernel_size: kernel size of conv layers
+        pool1_kernel_size: kernel size of first pooling layer
+        pool1_strides: strides of first pooling layer
+        pool2_kernel_size: kernel size of second pooling layer
+        pool2_strides: strides of second pooling layer
+        activation: activation function at the end
+    """
+
+    in_features: int
+    conv1_channels: int
+    conv2_channels: int
+    conv3_channels: int
+    conv4_channels: int
+    conv_kernel_size: Union[int, Tuple[int, ...]]
+    pool1_kernel_size: Union[int, Tuple[int, ...]]
+    pool1_strides: Optional[Union[int, Tuple[int, ...]]]
+    pool2_kernel_size: Union[int, Tuple[int, ...]]
+    pool2_strides: Optional[Union[int, Tuple[int, ...]]]
+    activation: Union[nn.Module, Callable[[torch.Tensor], torch.Tensor]]
+
+    def check_valid(self):
+        if isinstance(self.conv_kernel_size, int):
+            assert self.conv_kernel_size % 2 == 1, "ConformerVGGFrontendV1 only supports odd kernel sizes"
+        if isinstance(self.pool1_kernel_size, int):
+            assert self.pool1_kernel_size % 2 == 1, "ConformerVGGFrontendV1 only supports odd kernel sizes"
+        if isinstance(self.pool2_kernel_size, int):
+            assert self.pool2_kernel_size % 2 == 1, "ConformerVGGFrontendV1 only supports odd kernel sizes"
+
+    def __post__init__(self):
+        super().__post_init__()
+        self.check_valid()
+
+
+class ConformerVGGFrontendV1(nn.Module):
+    """
+    Convolutional Front-End
```

Comment:
As said above, you should describe with words (very shortly, maybe 5-10 words) what this does (so that the reader can easily grasp the differences to other frontends).

Also, where do you have this from? Some references would be nice. Did we use that in some other papers? Referencing those papers would also be helpful.